media-libs/amdgpu-pro-vulkan/amdgpu-pro-vulkan-22.10.4.1452060-r1.ebuild

```diff
@@ -115,20 +118,59 @@ src_install() {
 		insinto /usr/share/vulkan/icd.d
 		doins "${S}"/"${PN}"-i386/opt/amdgpu-pro/etc/vulkan/icd.d/amd_pro_icd32.json
 	fi
+
+	# AMD Vulkan Prefixes
+	newbin "${WORKDIR}/amd-vulkan-prefixes-main/amd_vulkan_prefixes.sh" vk_radv
+	newbin "${WORKDIR}/amd-vulkan-prefixes-main/amd_vulkan_prefixes.sh" vk_amdvlk
+	newbin "${WORKDIR}/amd-vulkan-prefixes-main/amd_vulkan_prefixes.sh" vk_pro
+
+	newbashcomp "${WORKDIR}/amd-vulkan-prefixes-main/amd_vulkan_prefixes.bash-completion" vk_radv
+	newbashcomp "${WORKDIR}/amd-vulkan-prefixes-main/amd_vulkan_prefixes.bash-completion" vk_amdvlk
+	newbashcomp "${WORKDIR}/amd-vulkan-prefixes-main/amd_vulkan_prefixes.bash-completion" vk_pro
+
+	if use use-as-default ; then
+		#set AMDGPU-PRO ICD loader system wide
+		doenvd "${FILESDIR}/99-amdgpu-pro-icd"
```

Comment:
Here we force to use the proprietary driver actually, that means other drivers won't even be found. :thinking: 

I believe the proper way is to offer an `eselect` approach instead of use flags, now that I think about it, what do you think ?

Reply:
Yha the eselect approach would be a good option just need to see how that could be done

Reply:
I think the better approach would be to create for exemple a new package eselect-vulkan-loader
After verify if the user have nouveau, nvidia, intel or amd the eselect in give option like this:

1 - IGPU - RADV (or Intel equivalent)
2 - IGPU+DGPU (intel+nvidia, amd+nvidia or amd+amd or intel+intel)
3 - DGPU (this is good for exemple laptop the have mux switch)

This is kinda of a rough ideia depending of the useflag for the video_card we could define this better.

Reply:
I think we need to look more into knowing how the default driver is chosen. Our goal is to change the default driver, not override the list :thinking: